xbmc/cores/VideoPlayer/DVDCodecs/Video/VAAPI.cpp

```diff
@@ -3068,8 +3068,10 @@ bool CFFmpegPostproc::AddPicture(CVaapiDecodedPicture &inPic)
   m_pFilterFrameIn->height = m_config.vidHeight;
   m_pFilterFrameIn->linesize[0] = image.pitches[0];
   m_pFilterFrameIn->linesize[1] = image.pitches[1];
-  m_pFilterFrameIn->interlaced_frame = (inPic.DVDPic.iFlags & DVP_FLAG_INTERLACED) ? 1 : 0;
-  m_pFilterFrameIn->top_field_first = (inPic.DVDPic.iFlags & DVP_FLAG_TOP_FIELD_FIRST) ? 1 : 0;
+  if (inPic.DVDPic.iFlags & DVP_FLAG_INTERLACED)
+    m_pFilterFrameIn->flags &= AV_FRAME_FLAG_INTERLACED;
```

Comment:
|= else you most likely set all flags to 0. As the & condition gets "0" and that result you AND on top.

Reply:
👍 Haven't noticed this code path at first since it's not used on Mac so it's untested. But your suggestion is right. There's also another deprecation inside `VAAPI.cpp`:
```C++
  m_pFilterFrameIn->pkt_size = image.data_size;
```
The docs say to set `AV_CODEC_FLAG_COPY_OPAQUE` and then pass data through `opaque` and `opaque_ref` but I'm not sure how that applies here.